src/Microsoft.ML.Data/Data/IColumn.cs

```diff
@@ -124,14 +124,13 @@ private static IColumn GetColumnCore<T>(ISchema schema, int col)
         /// <summary>
         /// Wraps the metadata of a column as a row.
         /// </summary>
-        /// <param name="schema"></param>
-        /// <param name="col"></param>
-        /// <returns></returns>
-        public static IRow GetMetadataAsRow(ISchema schema, int col)
+        public static IRow GetMetadataAsRow(ISchema schema, int col, Func<string, bool> takeMetadata)
```

Comment:
>takeMetadata [](start = 88, length = 12)

I understand idea, but I don't see any place where you actually use it.
Is it for future PRs?

Reply:
No, it's here. Check line 174 in `KeyToValue`

---
In reply to: [216508598](https://github.com/dotnet/machinelearning/pull/856#discussion_r216508598) [](ancestors = 216508598)

Reply:
Well hidden reference, ok you need it, but I still don't like what we need to change code in multiple places in order to make it comply with CheckValue(takeMetadata)
Can this one be nullable option?


---
In reply to: [216509221](https://github.com/dotnet/machinelearning/pull/856#discussion_r216509221) [](ancestors = 216509221,216508598)